### PR TITLE
Update bazel workspace to latest versions.

### PR DESCRIPTION
**Pull request by jverkoey**
This increases the following versions:

- bazel from 0.11 to 0.20.0
- build_bazel_rules_apple from 7ea0557 to 0.9.0
- build_bazel_rules_swift to 0.4.0 (new)
- bazel_ios_warnings from v1.0.1 to v2.0.0
- Xcode from 8 to 9.
- Swift pinned to 3.
- Unit test upgraded to UI tests.
- MotionInterchange from v1.6.0 to v2.0.0
- Migrate from the bazel runner scripts to the explicit Xcode selection + run pattern used by material-components-ios (https://github.com/material-components/material-components-ios/blob/fe0099d65c4fa67a02d7b842baf16b540bb2fa86/.kokoro#L102)

Tested by running:

    ./.kokoro

### CODE REVIEW

File: .kokoro [hunk 1]
```diff
@@ -20,53 +20,69 @@ set -e
 # Display commands to stderr.
 set -x
 
-KOKORO_RUNNER_VERSION="v4.*"
```
Comment (romoore) [inline .kokoro:23]: Removing the KOKORO runner?

File: .kokoro [hunk 2]
```diff
@@ -20,53 +20,69 @@ set -e
...
-  bazel version
-fi
-
-./.kokoro-ios-runner/bazel.sh test //:UnitTests --min-xcode-version 8.1.0 $verbosity_args
+BAZEL_VERSION="0.20.0"
+
+# xcode-select's the provided xcode version.
+# Usage example:
+#     select_xcode 9.2.0
+select_xcode() {
```
Comment (romoore) [inline .kokoro]: This looks like a significant refactor/rewrite. Could you please update the PR summary to reflect what's being done in this PR?

File: .kokoro [hunk 3]
```diff
@@ -20,53 +20,69 @@ set -e
...
+  desired_version="$1"
+  if [ -z "$desired_version" ]; then
+    return # No Xcode version to select.
```
Comment (romoore) [inline .kokoro]: Should this be an error? We tried to select an xcode and couldn't?

File: .kokoro [hunk 4]
```diff
@@ -20,53 +20,69 @@ set -e
...
+  fi
+
+  xcodes=$(ls /Applications/ | grep "Xcode")
+  for xcode_path in $xcodes; do
+    xcode_version=$(cat /Applications/$xcode_path/Contents/version.plist \
+      | grep "CFBundleShortVersionString" -A1 \
+      | grep string \
+      | cut -d'>' -f2 \
+      | cut -d'<' -f1)
+    xcode_version_as_number="$(version_as_number $xcode_version)"
```
Comment (romoore) [inline .kokoro]: Where is version_as_number defined? I don't see it declared before this method.

File: .kokoro [hunk 5]
```diff
@@ -20,53 +20,69 @@ set -e
...
+
+  # Run against whichever Xcode is currently selected.
+  selected_xcode_developer_path=$(xcode-select -p)
+  selected_xcode_contents_path=$(dirname "$selected_xcode_developer_path")
+
+  xcode_version=$(cat "$selected_xcode_contents_path/version.plist" \
+    | grep "CFBundleShortVersionString" -A1 \
+    | grep string \
+    | cut -d'>' -f2 \
+    | cut -d'<' -f1)
```
Comment (romoore) [inline .kokoro]: Tip: This chain of `grep` and `cut` commands look like it could be defined in a function and reused.

File: .kokoro [hunk 6]
```diff
@@ -20,53 +20,69 @@ set -e
...
+
+  bazel clean
+  bazel test //... --xcode_version $xcode_version --ios_minimum_os=8.0 --ios_multi_cpus=i386,x86_64
```
Comment (romoore) [inline .kokoro]: Tip: Make `ios_minimum_os` a variable in the script.

**Review by romoore — APPROVED** [review]
Please update the PR summary to document the `.kokoro` script refactoring changes as well.

File: .kokoro (hunk 1 above)
Comment (jverkoey) [reply .kokoro:23]: Yep - the runner's going away because it runs "all of the xcodes" which causes a variety of problems.

File: .kokoro (hunk 2 above)
Comment (jverkoey) [reply .kokoro]: Done.

File: .kokoro (hunk 3 above)
Comment (jverkoey) [reply .kokoro]: In this case we just use the currently select Xcode, which is the desired behavior.

File: .kokoro (hunk 1 above)
Comment (romoore) [reply .kokoro:23]: Great, thanks for updating!

File: .kokoro (hunk 5 above)
Comment (jverkoey) [reply .kokoro]: Done.

File: .kokoro (hunk 6 above)
Comment (jverkoey) [reply .kokoro]: Done.

File: .kokoro (hunk 4 above)
Comment (jverkoey) [reply .kokoro]: I've deleted this method because it wasn't being used.

**Pull request merged**